ui/apps/platform/src/Containers/Collections/CollectionResults.tsx

```diff
@@ -1,7 +1,153 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState, useMemo } from 'react';
+import {
+    Button,
+    Divider,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Flex,
+    FlexItem,
+    Select,
+    SearchInput,
+    SelectOption,
+    debounce,
+} from '@patternfly/react-core';
+import { ListIcon } from '@patternfly/react-icons';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import ResourceIcon from 'Components/PatternFly/ResourceIcon';
 
-function CollectionResults() {
-    return <></>;
+import { dryRunCollection } from 'services/CollectionsService';
+import { ListDeployment } from 'types/deployment.proto';
+
+function CollectionResults({ dryRunConfig }) {
+    const { isOpen, onToggle, closeSelect } = useSelectToggle();
+    const [selected, setSelected] = useState('Deployment');
+    const [filterText, setFilterText] = useState('');
+    const [isEndOfResults, setIsEndOfResults] = useState(false);
+    const [deployments, setDeployments] = useState<ListDeployment[]>([]);
+    const [hasRuleSelected, setHasRuleSelected] = useState(false);
+
+    function onRuleOptionSelect(_, value) {
+        setSelected(value);
+        setFilterText('');
+        closeSelect();
+    }
+
+    const fetchDryRun = useCallback(
+        (page: number) => {
+            const pageSize = 10;
+            const query = filterText ? `${selected}:${filterText}` : '';
```

Comment:
There is a `getRequestQueryStringForSearchFilter` util function that will create the search string for you that would come in handy if/when we apply multiple searches to this section.